tests/e2e/asserts_test.go

```diff
@@ -2281,10 +2281,16 @@ func OfflineResizePVC(namespace, clusterName string, timeout int) {
 				// Deleting standby and replica pods
 				err = env.DeletePod(namespace, pod.Name, forceDelete)
 				Expect(err).ToNot(HaveOccurred())
-				// Ensuring cluster is healthy with three pods
-				AssertClusterIsReady(namespace, clusterName, timeout, env)
 			}
 		}
+		// Make sure the cluster phase changed first
+		AssertClusterEventuallyDepartsPhase(namespace, clusterName, apiv1.PhaseHealthy, timeout)
+		// Ensuring cluster is healthy with three pods, we only need verify here once to safe time,
+		// but we need to make sure the cluster changed to un-health first, otherwise, it will be too
+		// quick, the status of the cluster did not have time to change from healthy to un-health, but
+		// the code reach to deleting primary pod.
+		AssertClusterIsReady(namespace, clusterName, timeout, env)
```

Comment:
checking for cluster ready after cluster is no longer healthy (after pods have been deleted) is surprising me

Reply:
I think we should do it differently. AssertClusterIsReady should make sure that none of the Cluster's pods is deleted.